File: Data Structures & Algorithms/missing-number/submission-3.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = 0;
        int i = 0;
        for(i ; i < nums.size(); i++){
            res ^= nums[i];
            res ^= i;
        }
        res ^= i;

        return res;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums)
        mul = 1
        zero_count = 0
        for num in nums:
            if num:
                mul *= num
            else:
                zero_count +=1
        if zero_count > 1:
            return [0]* len(nums)
        for i,num in enumerate(nums):
            if zero_count > 0:
                if num == 0:
                    res[i] = mul
                else:
                    res[i] = 0
            else:
                res[i] = mul // num
        return res
